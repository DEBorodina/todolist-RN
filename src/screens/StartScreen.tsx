import React from 'react';
import { Text, View } from 'react-native';

export const StartScreen = () => {
  return (
    <View>
      <Text>StartScreen</Text>
    </View>
  );
};
