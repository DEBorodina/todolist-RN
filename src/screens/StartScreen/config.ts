export const BUTTON_TEXT = 'Get started';
export const TITLE = 'Manage your tasks';
export const SUBTITLE =
  'Organize, plan, and collaborate on tasks with Modsen todo list.Your busy life deserves this.you can manage checklist and your goal.';

export const ANIMATION_DURATION = 700;
