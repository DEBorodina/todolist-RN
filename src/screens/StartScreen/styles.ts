import styled from 'styled-components/native';

export const Container = styled.View`
  width: 80%;
  align-items: center;
`;

export const Layout = styled.View`
  width: 100%;
  max-width: 600px;
  height: 80%;
  max-height: 550px;
  align-items: center;
  justify-content: space-between;
  margin: auto;
`;
