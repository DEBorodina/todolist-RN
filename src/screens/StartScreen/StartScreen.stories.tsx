import type { Meta } from '@storybook/react';

import { StartScreen } from './StartScreen';

const StartScreenMeta: Meta<typeof StartScreen> = {
  title: 'StartScreen',
  component: StartScreen,
};

export default StartScreenMeta;
