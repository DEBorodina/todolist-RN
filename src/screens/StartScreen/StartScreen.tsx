import React, { FC } from 'react';
import { Image, Text, View } from 'react-native';

import { styles } from './styles';
import { StartScreenProps } from './types';

export const StartScreen: FC<StartScreenProps> = () => (
  <View style={styles.container}>
    <Image source={require('../../../assets/images/start-background.png')} />
    <Text>StartScreen 3</Text>
  </View>
);
