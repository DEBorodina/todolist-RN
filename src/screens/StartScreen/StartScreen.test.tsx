import React from 'react';

import { render } from '@test-utils';

import { StartScreen } from './StartScreen';

// нужно замокать async storage и редакс
describe('StartScreen', () => {
  it('renders correctly', () => {
    const component = render(<StartScreen />);
    expect(component).toMatchSnapshot();
  });
});
