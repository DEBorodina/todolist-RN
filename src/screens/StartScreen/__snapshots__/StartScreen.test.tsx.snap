// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartScreen renders correctly 1`] = `
<View
  style={{}}
>
  <Image
    source={
      {
        "testUri": "../../../assets/images/start-background.png",
      }
    }
  />
  <Text>
    StartScreen 3
  </Text>
</View>
`;
